### PR TITLE
Ver 96590 update developer doc

### PR DESCRIPTION
**Pull request by LiboYu2**
Read the DEVELOPER.md. Updated it with latest changes and fixed some typos and errors.
Updated docker-build target. Now docker-build-vertica-v2 will be triggered. The v2 includes node management agent while v1 does not have it.

### CODE REVIEW

File: DEVELOPER.md [hunk 1]
```diff
@@ -333,7 +337,7 @@ Unit tests for the VerticaDB operator use the Go testing infrastructure. Some te
 ## e2e Tests
 
 > **IMPORTANT**
-> The e2e tests only run on operators that were [deployed as an object](#deployment-object).
+> The e2e tests only run on operators that were deployed as an object.
```
Comment (qindotguan) [inline DEVELOPER.md:340]: This line should be removed. Deploy as an object is the only deployment option mentioned in this document (see this commit: https://github.com/vertica/vertica-kubernetes/commit/04390c61a1c811749a02b67b38d11d4c99bcf764).

Comment (LiboYu2) [reply DEVELOPER.md:340]: Will do

File: DEVELOPER.md [hunk 2]
```diff
@@ -179,17 +179,17 @@ The following steps build the images and push them to the Kind cluster in the cu
...
+   vertica-k8s          2.2.0     46e4511cabf1   5 days ago       1.62GB
+   rockylinux           9         639282825872   2 weeks ago      70.3MB
    ...
    ```
 
    - `vertica-k8s`: long-running container that runs the Vertica daemon. This container is designed for admintools deployments. For details about the admintools deployment image, see the [Dockerfile](./docker-vertica/Dockerfile). For details about the vcluster deployment image, see the [Dockerfile](./docker-vertica-v2/Dockerfile).
    - `verticadb-operator`: runs the VerticaDB operator and webhook. For details, see the [Dockerfile](./docker-operator/Dockerfile).
    - `vertica-logger`: runs the vlogger sidecar container that sends the contents of `vertica.log` to STDOUT. For details, see the [Dockerfile](./docker-vlogger/Dockerfile).
-   - `ubuntu`: serves as the base image for the `vertica-k8s` image. The `make docker-build` command pulls the latest version each time.
+   - `rockylinux: serves as the base image for the `vertica-k8s` image. The `make docker-build` command pulls the latest version each time.
```
Comment (cchen-vertica) [inline DEVELOPER.md:192]:
```suggestion
   - `rockylinux9: serves as the default base image for the `vertica-k8s` image. The `make docker-build` command pulls this image each time.
```

Comment (LiboYu2) [reply DEVELOPER.md]: Added the update.

**Review by roypaulin — APPROVED** [review]

**Pull request merged**